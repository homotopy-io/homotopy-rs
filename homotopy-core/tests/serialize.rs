use homotopy_core::{serialize::*, *};
use insta::*;

#[test]
fn serialize_associator() {
    let (_, diagram) = examples::associator();

    let (serialized, key) = {
        let mut store = Store::new();
        let key = store.pack_diagram(&diagram.clone().into());
        let serialized = rmp_serde::encode::to_vec_named(&store).unwrap();
        (serialized, key)
    };

    assert_debug_snapshot!(base64::encode(&serialized));

    let deserialized = {
        let mut store: Store = rmp_serde::decode::from_slice(&serialized).unwrap();
        store.unpack_diagram(key).unwrap()
    };

    assert_eq!(Diagram::from(diagram), deserialized);
}
