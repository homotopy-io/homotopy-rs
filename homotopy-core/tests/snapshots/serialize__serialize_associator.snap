---
source: homotopy-core/tests/serialize.rs
expression: "base64::encode(&serialized)"

---
"g6hkaWFncmFtc4SSzyj93Ib9LnI0zyLAuLnpNbeNhKR0eXBlokRuqWRpbWVuc2lvbgKmc291cmNlks/ww9dwoIO58s8RUA/MjY3QXqdjb3NwYW5zkoKnZm9yd2FyZJLPQgSMKYOCDpLPn0/shfusLNWoYmFja3dhcmSSz/17Rv1POS8MzwpdjgWtYu9Pgqdmb3J3YXJkks9CBIwpg4IOks+fT+yF+6ws1ahiYWNrd2FyZJLP/XtG/U85LwzPCl2OBa1i70+Sz2qJBF8K1SFaz22bEac4uFR5gqR0eXBlokQwqWdlbmVyYXRvcoKpZGltZW5zaW9uAKJpZACSz8szkgH4rfeez5yiWFmwuFnehKR0eXBlokRuqWRpbWVuc2lvbgOmc291cmNlks8o/dyG/S5yNM8iwLi56TW3jadjb3NwYW5zkYKnZm9yd2FyZJLPGSujdCWnbdfPB7+Klg+elC6oYmFja3dhcmSSz/pjRlUiXtExz9D+v/l1hpG8ks/ww9dwoIO58s8RUA/MjY3QXoSkdHlwZaJEbqlkaW1lbnNpb24BpnNvdXJjZZLPaokEXwrVIVrPbZsRpzi4VHmnY29zcGFuc5OCp2ZvcndhcmSSz6uKyfLEUXWXz+kqdaLkjGBbqGJhY2t3YXJkks9Vy9FCB8Kslc8s9w3iF7WWv4KnZm9yd2FyZJLPq4rJ8sRRdZfP6Sp1ouSMYFuoYmFja3dhcmSSz1XL0UIHwqyVzyz3DeIXtZa/gqdmb3J3YXJkks+risnyxFF1l8/pKnWi5IxgW6hiYWNrd2FyZJLPVcvRQgfCrJXPLPcN4he1lr+ocmV3cml0ZXPeABiSzw39GhEO3Ed4z4YLPpT8Lxvzg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLPFW4DbIXT9X3P0h0/siDEybSSzxkro3Qlp23Xzwe/ipYPnpQug6R0eXBlolJuqWRpbWVuc2lvbgKlY29uZXORgqVpbmRleACkY29uZZLPmOngZTsKxDTPQUZUIRZB8i+Sz0IEjCmDgg6Sz59P7IX7rCzVg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLP2hMudZY9jrLPkdyMzvumoGOSz1XL0UIHwqyVzyz3DeIXtZa/hKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAKJpZACmdGFyZ2V0gqlkaW1lbnNpb24BomlkAaVsYWJlbJKCqWRpbWVuc2lvbgGiaWQBkpECkQGSz1cRPbhJTaK7z7icMbHUwJW5hKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAKJpZACmdGFyZ2V0gqlkaW1lbnNpb24DomlkA6VsYWJlbJKCqWRpbWVuc2lvbgOiaWQDkpECkQGSz1dd0eZDDEDPzyWmkRnSje/3hKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAKJpZACmdGFyZ2V0gqlkaW1lbnNpb24ComlkAqVsYWJlbJKCqWRpbWVuc2lvbgKiaWQCkpEAkQGSz2NEJKv8Mkv+z8hGGMlaRyK7hKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAaJpZAGmdGFyZ2V0gqlkaW1lbnNpb24DomlkA6VsYWJlbJKCqWRpbWVuc2lvbgOiaWQDkpECkQGSz3JmFA2V907Yz14BEmVYG1rBhKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAqJpZAKmdGFyZ2V0gqlkaW1lbnNpb24DomlkA6VsYWJlbJKCqWRpbWVuc2lvbgOiaWQDkpEAkQGSz3LaVhOB8615z7EFmezeQuCOhKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAaJpZAGmdGFyZ2V0gqlkaW1lbnNpb24ComlkAqVsYWJlbJKCqWRpbWVuc2lvbgKiaWQCkpEAkQGSz3Y+UiAtZ+30z8J+6FEHF8wZhKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAqJpZAKmdGFyZ2V0gqlkaW1lbnNpb24DomlkA6VsYWJlbJKCqWRpbWVuc2lvbgOiaWQDkpECkQGSz3qSqyFH/55Jz/Krur121FCag6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLPHLf+styH2VLPywPZgTR/soySz55oFSNiL8j7z+XR8ndugDOehKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAaJpZAGmdGFyZ2V0gqlkaW1lbnNpb24DomlkA6VsYWJlbJKCqWRpbWVuc2lvbgOiaWQDkpEAkQGSz6KQX1x6Nwy+z+DrPCBFWOrYg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLPwu63D1Mtm9fPaF5ZGFUc7vqSz6uKyfLEUXWXz+kqdaLkjGBbhKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAKJpZACmdGFyZ2V0gqlkaW1lbnNpb24BomlkAaVsYWJlbJKCqWRpbWVuc2lvbgGiaWQBkpEAkQGSz7H/NCyh7FN9zxGvVZKSqrZwhKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAaJpZAGmdGFyZ2V0gqlkaW1lbnNpb24ComlkAqVsYWJlbJKCqWRpbWVuc2lvbgKiaWQCkpECkQGSz8AwtkDI0XTpzwqQZVveYSlRg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleAGkY29uZZLP2hMudZY9jrLPkdyMzvumoGOSz9PBxrDM0R1Tz4SuoLRvr5LxhKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAKJpZACmdGFyZ2V0gqlkaW1lbnNpb24DomlkA6VsYWJlbJKCqWRpbWVuc2lvbgOiaWQDkpEAkQGSz+u/csu07U1Az5JrAY1LBlBtg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLPhN7HFt0dtcDPFvtQcdAgKOaSz/Lsw98PJD3+z1ggzlDFnkvYg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLPcWpHMdAKAl3P3bMuZftJ5BiSz/eChxfIPz3pz7X1jltHGWt0g6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLPs/tS2o1bhRLPDFWlP4qw/g+Sz/iFt/uQMdwMz8NlW747LAZ4hKR0eXBlolIwpnNvdXJjZYKpZGltZW5zaW9uAKJpZACmdGFyZ2V0gqlkaW1lbnNpb24ComlkAqVsYWJlbJKCqWRpbWVuc2lvbgKiaWQCkpECkQGSz/pjRlUiXtExz9D+v/l1hpG8g6R0eXBlolJuqWRpbWVuc2lvbgKlY29uZXORgqVpbmRleACkY29uZZLP7cnb+KvXdcHPeakOTlxIO7iSz/yx3IBpgg9Jz1xAs6pLhT4lg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleAGkY29uZZLP4U/gQG8GdmHPxzVG7OZgdvWSz/17Rv1POS8MzwpdjgWtYu9Pg6R0eXBlolJuqWRpbWVuc2lvbgGlY29uZXORgqVpbmRleACkY29uZZLP4U/gQG8GdmHPxzVG7OZgdvWlY29uZXOKks8VbgNshdP1fc/SHT+yIMTJtIOmc291cmNlkYKnZm9yd2FyZJLPq4rJ8sRRdZfP6Sp1ouSMYFuoYmFja3dhcmSSz1XL0UIHwqyVzyz3DeIXtZa/pnRhcmdldIKnZm9yd2FyZJLP08HGsMzRHVPPhK6gtG+vkvGoYmFja3dhcmSSz1cRPbhJTaK7z7icMbHUwJW5pnNsaWNlc5GSz2NEJKv8Mkv+z8hGGMlaRyK7ks8ct/6y3IfZUs/LA9mBNH+yjIOmc291cmNlkYKnZm9yd2FyZJLPV13R5kMMQM/PJaaRGdKN7/eoYmFja3dhcmSSz/iFt/uQMdwMz8NlW747LAZ4pnRhcmdldIKnZm9yd2FyZJLP08HGsMzRHVPPhK6gtG+vkvGoYmFja3dhcmSSz1cRPbhJTaK7z7icMbHUwJW5pnNsaWNlc5GSz3Y+UiAtZ+30z8J+6FEHF8wZks9xakcx0AoCXc/dsy5l+0nkGIOmc291cmNlkoKnZm9yd2FyZJLPq4rJ8sRRdZfP6Sp1ouSMYFuoYmFja3dhcmSSz1XL0UIHwqyVzyz3DeIXtZa/gqdmb3J3YXJkks9XXdHmQwxAz88lppEZ0o3v96hiYWNrd2FyZJLP+IW3+5Ax3AzPw2VbvjssBnimdGFyZ2V0gqdmb3J3YXJkks/TwcawzNEdU8+ErqC0b6+S8ahiYWNrd2FyZJLPVxE9uElNorvPuJwxsdTAlbmmc2xpY2VzkpLPY0Qkq/wyS/7PyEYYyVpHIruSz3Y+UiAtZ+30z8J+6FEHF8wZks+E3scW3R21wM8W+1Bx0CAo5oOmc291cmNlk4KnZm9yd2FyZJLPq4rJ8sRRdZfP6Sp1ouSMYFuoYmFja3dhcmSSz1XL0UIHwqyVzyz3DeIXtZa/gqdmb3J3YXJkks+risnyxFF1l8/pKnWi5IxgW6hiYWNrd2FyZJLPVcvRQgfCrJXPLPcN4he1lr+Cp2ZvcndhcmSSz6uKyfLEUXWXz+kqdaLkjGBbqGJhY2t3YXJkks9Vy9FCB8Kslc8s9w3iF7WWv6Z0YXJnZXSCp2ZvcndhcmSSz9PBxrDM0R1Tz4SuoLRvr5LxqGJhY2t3YXJkks9XET24SU2iu8+4nDGx1MCVuaZzbGljZXOTks+eaBUjYi/I+8/l0fJ3boAznpLPnmgVI2IvyPvP5dHyd26AM56Sz55oFSNiL8j7z+XR8ndugDOeks+Y6eBlOwrENM9BRlQhFkHyL4Omc291cmNlkoKnZm9yd2FyZJLPQgSMKYOCDpLPn0/shfusLNWoYmFja3dhcmSSz/17Rv1POS8MzwpdjgWtYu9Pgqdmb3J3YXJkks9CBIwpg4IOks+fT+yF+6ws1ahiYWNrd2FyZJLP/XtG/U85LwzPCl2OBa1i70+mdGFyZ2V0gqdmb3J3YXJkks/rv3LLtO1NQM+SawGNSwZQbahiYWNrd2FyZJLPDf0aEQ7cR3jPhgs+lPwvG/Omc2xpY2VzkpLP94KHF8g/PenPtfWOW0cZa3SSz6KQX1x6Nwy+z+DrPCBFWOrYks+z+1LajVuFEs8MVaU/irD+D4Omc291cmNlkoKnZm9yd2FyZJLPV13R5kMMQM/PJaaRGdKN7/eoYmFja3dhcmSSz/iFt/uQMdwMz8NlW747LAZ4gqdmb3J3YXJkks+risnyxFF1l8/pKnWi5IxgW6hiYWNrd2FyZJLPVcvRQgfCrJXPLPcN4he1lr+mdGFyZ2V0gqdmb3J3YXJkks/TwcawzNEdU8+ErqC0b6+S8ahiYWNrd2FyZJLPVxE9uElNorvPuJwxsdTAlbmmc2xpY2VzkpLPcmYUDZX3TtjPXgESZVgbWsGSz55oFSNiL8j7z+XR8ndugDOeks/C7rcPUy2b189oXlkYVRzu+oOmc291cmNlkYKnZm9yd2FyZJLPV13R5kMMQM/PJaaRGdKN7/eoYmFja3dhcmSSz/iFt/uQMdwMz8NlW747LAZ4pnRhcmdldIKnZm9yd2FyZJLP08HGsMzRHVPPhK6gtG+vkvGoYmFja3dhcmSSz1cRPbhJTaK7z7icMbHUwJW5pnNsaWNlc5GSz3JmFA2V907Yz14BEmVYG1rBks/aEy51lj2Oss+R3IzO+6agY4Omc291cmNlkoKnZm9yd2FyZJLPq4rJ8sRRdZfP6Sp1ouSMYFuoYmFja3dhcmSSz1XL0UIHwqyVzyz3DeIXtZa/gqdmb3J3YXJkks+risnyxFF1l8/pKnWi5IxgW6hiYWNrd2FyZJLPVcvRQgfCrJXPLPcN4he1lr+mdGFyZ2V0gqdmb3J3YXJkks9XXdHmQwxAz88lppEZ0o3v96hiYWNrd2FyZJLP+IW3+5Ax3AzPw2VbvjssBnimc2xpY2VzkpLPctpWE4HzrXnPsQWZ7N5C4I6Sz3LaVhOB8615z7EFmezeQuCOks/hT+BAbwZ2Yc/HNUbs5mB29YOmc291cmNlkYKnZm9yd2FyZJLPq4rJ8sRRdZfP6Sp1ouSMYFuoYmFja3dhcmSSz1XL0UIHwqyVzyz3DeIXtZa/pnRhcmdldIKnZm9yd2FyZJLPV13R5kMMQM/PJaaRGdKN7/eoYmFja3dhcmSSz/iFt/uQMdwMz8NlW747LAZ4pnNsaWNlc5GSz7H/NCyh7FN9zxGvVZKSqrZwks/tydv4q9d1wc95qQ5OXEg7uIOmc291cmNlkoKnZm9yd2FyZJLPwDC2QMjRdOnPCpBlW95hKVGoYmFja3dhcmSSz/yx3IBpgg9Jz1xAs6pLhT4lgqdmb3J3YXJkks9CBIwpg4IOks+fT+yF+6ws1ahiYWNrd2FyZJLP/XtG/U85LwzPCl2OBa1i70+mdGFyZ2V0gqdmb3J3YXJkks/rv3LLtO1NQM+SawGNSwZQbahiYWNrd2FyZJLPDf0aEQ7cR3jPhgs+lPwvG/Omc2xpY2VzkpLP8uzD3w8kPf7PWCDOUMWeS9iSz3qSqyFH/55Jz/Krur121FCa"
